src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -0,0 +1,317 @@
+        {
+            // Warn the user.
+            await ThreadHandling.Value.SwitchToUIThread();
+
+            // Check if already warned - this could happen in the off chance two projects are added very quickly since the detection work is 
+            // scheduled on idle.
+            if(_compatibilityLevelWarnedForThisSolution < compatLevel)
+            {
+                // Only want to warn once per solution
+                _compatibilityLevelWarnedForThisSolution = compatLevel;
+                
+                IVsUIShell uiShell = _serviceProvider.GetService<IVsUIShell, SVsUIShell>();
+                uiShell.GetAppName(out string caption);
+
+                if(compatLevel == CompatibilityLevel.Partial)
+                {
+                    // Get current dontShowAgain value
+                    var settingsManager = (ISettingsManager)_serviceProvider.GetService(typeof(SVsSettingsPersistenceManager));
+                    bool suppressPrompt = false;
+                    if( settingsManager != null)
+                    {
+                        suppressPrompt = settingsManager.GetValueOrDefault(SuppressDotNewCoreWarningKey, defaultValue:false);
+                    }
+                    
+                    if(!suppressPrompt)
+                    {
+                        suppressPrompt= DialogServices.Value.DontShowAgainMessageBox(caption, VSResources.PartialSupportedDotNetCoreProject, VSResources.DontShowAgain, false, VSResources.LearnMore, LearnMoreFwlink);
+                        if(suppressPrompt && settingsManager != null)
+                        {
+                            await settingsManager.SetValueAsync(SuppressDotNewCoreWarningKey, true, isMachineLocal: true).ConfigureAwait(true);
+                        }
+                    }
+                }
+                else
+                {
+                    DialogServices.Value.DontShowAgainMessageBox(caption, string.Format(VSResources.NotSupportedDotNetCoreProject, s_partialSupportedVersion.Major, s_partialSupportedVersion.Minor), 
+                                                                 null, false, VSResources.LearnMore, LearnMoreFwlink);
+                }
+            }
+        }
+
+        private async Task<CompatibilityLevel> GetProjectCompatibilityAsync(UnconfiguredProject project)
+        {
+            if (project.Capabilities.AppliesTo(ProjectCapability.CSharpOrVisualBasicOrFSharp))
+            {
+                IProjectProperties properties = project.Services.ActiveConfiguredProjectProvider.ActiveConfiguredProject.Services.ProjectPropertiesProvider.GetCommonProperties();
+                string tfm = await properties.GetEvaluatedPropertyValueAsync("TargetFrameworkMoniker").ConfigureAwait(false);
+                var fw = new FrameworkName(tfm);
+                if ((fw.Identifier.Equals(".NETCore", StringComparison.OrdinalIgnoreCase) || fw.Identifier.Equals(".NETCoreApp", StringComparison.OrdinalIgnoreCase)))
```

Comment:
Isn't `.NETCore` here for UWP apps, and isn't it already at 5 or something?

Reply:
Ah yes. I removed that.